types/block_finality.go

```diff
@@ -129,6 +133,26 @@ func (b *BlockNumberFinality) BlockNumber(ctx context.Context, requester ethereu
 	return b.Block.ApplyOffset(blockHeader.Number.Uint64(), b.Offset), nil
 }
 
+// Header gets the block header from RPC
+func (b *BlockNumberFinality) BlockHeader(
+	ctx context.Context,
+	requester ethereum.ChainReader,
+) (*types.Header, error) {
+	blockHeader, err := requester.HeaderByNumber(ctx, b.Block.toBigInt())
+	if err != nil {
+		log.Errorf("BlockNumberFinality.Header: Error getting block %s. Err: %s", b.String(), err.Error())
+		return nil, err
+	}
+
+	blockNumberOffset := b.Block.ApplyOffset(blockHeader.Number.Uint64(), b.Offset)
+	blockHeaderOffset, err := requester.HeaderByNumber(ctx, new(big.Int).SetUint64(blockNumberOffset))
+	if err != nil {
+		log.Errorf("BlockNumberFinality.Header: Error getting block %s. Err: %s", b.String(), err.Error())
+		return nil, err
+	}
+	return blockHeaderOffset, nil
```

Comment:
IMO we can simplify this function, by first invoking the `BlockNumber` (to resolve the actual block number) and then just returning `requester.HeaderByNumber`. WDYT?

```suggestion
	blockNum, err := b.BlockNumber(ctx, requester)
	if err != nil {
		return nil, fmt.Errorf("failed to resolve offseted block by number (block number: %s): %w", b, err)
	}

	return requester.HeaderByNumber(ctx, new(big.Int).SetUint64(blockNum))
```

Reply:
done: 66beb28ec8bc736d5067c31be3cacaffb8b2a48d